com.unity.netcode.gameobjects/Tests/Runtime/UniversalRpcTests.cs

```diff
@@ -1636,17 +1645,20 @@ public void TestMutualRecursion()
 
             VerifyNotReceived(NetworkManager.ServerClientId, s_ClientIds);
 
-            for (var i = 0; i < 10; ++i)
+            var clientListExpected = 1;
+            var serverListExpected = 2;
+            for (var i = 1; i <= 10; ++i)
             {
                 WaitForMessageReceivedWithTimeTravel<RpcMessage>(clientList);
-                VerifyRemoteReceived(NetworkManager.ServerClientId, NetworkManager.ServerClientId, nameof(UniversalRpcNetworkBehaviour.MutualRecursionClientRpc), clientIdArray, false, false);
+                VerifyRemoteReceived(NetworkManager.ServerClientId, NetworkManager.ServerClientId, nameof(UniversalRpcNetworkBehaviour.MutualRecursionClientRpc), clientIdArray, false, false, clientListExpected);
                 VerifyNotReceived(NetworkManager.ServerClientId, serverIdArray);
+                clientListExpected *= 2;
```

Comment:
This is surprising to me. I would have thought the received count would have incremented linearly, not exponentially. This might be indicative of another (unrelated) bug with mutual recursion.

Not related to your PR, but something I want to investigate.